fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/common/SchedulerJobHelper.java

```diff
@@ -115,6 +115,9 @@ public void executeJob(String jobName) throws InterruptedException {
 
                 // Executing Scheduler Job
                 runSchedulerJob(this.requestSpec, jobId.toString());
+                
+                // Need to give the job an opportunity to start - otherwise below might return false as job not yet started
+                Thread.sleep(15000);
```

Comment:
@ptuomola would you consider raising this change as a separate PR instead of as part of this one? Or we could try to make a real better fix, based on what you described in FINERACT-855...

Reply:
Of course - will squash the commits and remove this. Based on the emails, I believe Awasum was going to send a fix to FINERACT-855.

Reply:
@ptuomola, I am looking into it but I cannot seem to replicate the issue locally as when running ```./gradlew integrationTest --tests ClientSavingsIntegrationTest
``` Everything passes locally. Could it be that our Travis Vm is not powerful enough?
If you have time to get this done earlier than me. Do it.

My fix for FINERACT-855 involved doing something like this where we are accessing the jobHistoryData...like so:

``` Assert.assertEquals("Verifying Last Scheduler Job Status", "success",
                        jobHistoryData.get(jobHistoryData.size() == 0 ? jobHistoryData.size() : jobHistoryData.size() - 1).get("status"));``` 

So that we stop getting the ArrayOutOfBounceException.

That may not be a good solution.

Reply:
My suggestion would be that we wrap up this PR ASAP by strictly and naively :imp:  following the process outlined on https://github.com/apache/fineract#pull-requests ... @ptuomola just try to get this PR to fully pass the build by raising separate PRs, and then rebasing, for any tests that fail on you here. The "full" (proper) solution, whether `Thread.sleep(15000)` or something else, should IMHO be worked on completely separate from this PR.

Reply:
You are right @vorburger . lets just get his green